src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java

```diff
@@ -160,6 +164,12 @@ public ProvisionConfiguration updateIdentityConfiguration(ProvisionContext provi
                 writeCertificateAndKeyToPath(response, parameterMap.get(ROOT_PATH_PARAMETER_NAME).toString());
                 certificateOwnershipToken = response.certificateOwnershipToken;
             } else {
+                if (csrPrivateKeyPath == null) {
+                    throw new IllegalArgumentException(String.format(MISSING_REQUIRED_PARAMETERS_ERROR_FORMAT,
+                            CSR_PRIVATE_KEY_PATH_PARAMETER_NAME));
+                }
+                movePrivateKey(parameterMap.get(ROOT_PATH_PARAMETER_NAME).toString(), csrPrivateKeyPath);
```

Comment:
don't do this yet. Only copy the file when we succeed.

Reply:
so, the idea of success in this case is receipt of a successful CreateCertificateFromCsrResponse or RegisterThingResponse?

Reply:
This change has "csr" in the title, so which of those does this change apply to?

Reply:
CreateCertificateFromCsrResponse it is!